src/test/kotlin/no/nav/syfo/identhendelse/IdenthendelseServiceSpek.kt

```diff
@@ -0,0 +1,132 @@
+package no.nav.syfo.identhendelse
+
+import io.ktor.server.testing.*
+import kotlinx.coroutines.*
+import no.nav.syfo.client.azuread.AzureAdClient
+import no.nav.syfo.client.pdl.PdlClient
+import no.nav.syfo.pengestopp.Arsak
+import no.nav.syfo.pengestopp.SykepengestoppArsak
+import no.nav.syfo.pengestopp.database.addStatus
+import no.nav.syfo.pengestopp.database.getActiveFlags
+import no.nav.syfo.testutils.generator.generateKafkaIdenthendelseDTO
+import no.nav.syfo.testutils.*
+import no.nav.syfo.testutils.generator.generateStatusEndringer
+import org.amshove.kluent.internal.assertFailsWith
+import org.amshove.kluent.shouldBeEqualTo
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+object IdenthendelseServiceSpek : Spek({
+
+    describe(IdenthendelseServiceSpek::class.java.simpleName) {
+
+        with(TestApplicationEngine()) {
+            start()
+
+            val externalMockEnvironment = ExternalMockEnvironment()
+            val database = externalMockEnvironment.database
+            val azureAdClient = AzureAdClient(
+                azureAppClientId = externalMockEnvironment.environment.azureAppClientId,
+                azureAppClientSecret = externalMockEnvironment.environment.azureAppClientSecret,
+                azureTokenEndpoint = externalMockEnvironment.environment.azureTokenEndpoint,
+            )
+            val pdlClient = PdlClient(
+                azureAdClient = azureAdClient,
+                pdlClientId = externalMockEnvironment.environment.pdlClientId,
+                pdlUrl = externalMockEnvironment.environment.pdlUrl,
+            )
+
+            val identhendelseService = IdenthendelseService(
+                database = database,
+                pdlClient = pdlClient,
+            )
+
+            beforeGroup {
+                externalMockEnvironment.startExternalMocks()
+            }
+
+            afterGroup {
+                externalMockEnvironment.stopExternalMocks()
+            }
+
+            afterEachTest {
+                database.connection.dropData()
+            }
+
+            describe("Happy path") {
+                it("Skal oppdatere statusendring når person har fått ny ident") {
+                    val kafkaIdenthendelseDTO = generateKafkaIdenthendelseDTO(hasOldPersonident = true)
+                    val newIdent = kafkaIdenthendelseDTO.getActivePersonident()!!
+                    val oldIdent = kafkaIdenthendelseDTO.getInactivePersonidenter().first()
+
+                    val arsakList = listOf(
+                        Arsak(type = SykepengestoppArsak.BESTRIDELSE_SYKMELDING),
+                        Arsak(type = SykepengestoppArsak.AKTIVITETSKRAV)
```

Comment:
Det er bare pirk, men det er egentlig bare unntak som kan ha flere årsaker i listen, så dette skal vel ikke gå an 🤷

Reply:
Okey, men det er good to know, jeg kan jo bare fjerne det 👌🏼